airflow/providers/imap/hooks/imap.py

```diff
@@ -41,7 +42,7 @@ class ImapHook(BaseHook):
     :type imap_conn_id: str
     """
 
-    def __init__(self, imap_conn_id='imap_default'):
+    def __init__(self, imap_conn_id: str = 'imap_default') -> None:
         super().__init__()
         self.imap_conn_id = imap_conn_id
         self.mail_client = None
```

Comment:
```suggestion
        self.mail_client: Optional[imaplib.IMAP4_SSL] = None
```

Reply:
This still gives the same mypy errors.

Reply:
You need to add runtime-checks for type: See: https://github.com/apache/airflow/pull/9733
```python
if not self.cli:
    raise Exception("The 'cli' should be initialized before!")
```